src/main/java/Main.java

```diff
@@ -130,30 +136,33 @@ public static void main(String[] args) throws Exception {
         try {
             String v = System.getProperty("java.class.version");
             if (v!=null) {
+                String classVersionString = v.split("\\.")[0];
                 try {
-                    float javaVersion = Float.parseFloat(v);
+                    int javaVersion = Integer.parseInt(classVersionString);
                     //TODO: add support of version ranges instead of equality check
-                    if (javaVersion == REQUIRED_JAVA_VERSION) {
+                    if (SUPPORTED_JAVA_CLASS_VERSIONS.contains(javaVersion)) {
```

Comment:
Can we please, for instance, extract this logic into some `isJavaRuntimeVersionSupported()` method, and write tests for it?

Reply:
done